yt/yt/tests/integration/queries/test_mock.py

```diff
@@ -202,6 +205,44 @@ def test_rows_limit(self, query_tracker):
         assert not q.get_result(1)["is_truncated"]
 
 
+class TestQueryTrackerBan(YTEnvSetup):
+    NUM_QUERY_TRACKER = 1
+    DELTA_DRIVER_CONFIG = {
+        "cluster_connection_dynamic_config_policy": "from_cluster_directory",
+    }
+    QUERY_TRACKER_DYNAMIC_CONFIG = {"state_check_period": 2000}
+
+    def query_fails():
+        start_query("mock", "run_forever")
+        return False
+
+    @authors("mpereskokova")
+    def test_query_tracker_ban(self, query_tracker):
+        addresses = query_tracker.query_tracker.addresses
+        set(f"//sys/query_tracker/instances/{addresses[0]}/@banned", True)
+
+        with raises_yt_error() as err:
+            wait(TestQueryTrackerBan.query_fails)
+        assert err[0].contains_text("No alive peers found")
+        time.sleep(3)  # NB(mpereskokova): Sleep for multiple active_query_acquisition_periods, to ensure, that no new queries will be acquired
```

Comment:
I strongly object writing tests with sleep.
Let's export StateChecker state via orchid, and wait until the banned flag is updated from the cypress.